From d4978c413b5efcc8d2d3b0ca20ddcec8a8423e8c Mon Sep 17 00:00:00 2001
From: Alice Carlotti <alice.carlotti@arm.com>
Date: Wed, 7 Jan 2026 18:55:46 +0000
Subject: [PATCH] aarch64: Accept hyphenated extensions in --with-arch
 [PR123460]

Add "-" to the list of valid characters in an extension name that is
used when removing the first extension from an extension string.

gcc/ChangeLog:

	PR target/123460
	* config.gcc: Accept hyphens in aarch64 --with-arch extensions.
---
 gcc/config.gcc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config.gcc b/gcc/config.gcc
index 087aaa7be371..53be4a7cfe75 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -4304,7 +4304,7 @@ case "${target}" in
 				  echo "Unknown extension used in --with-$which=$val" 1>&2
 				  exit 1
 				fi
-				ext_val=`echo $ext_val | sed -E -e 's/[a-z0-9]+//'`
+				ext_val=`echo $ext_val | sed -E -e 's/[a-z0-9-]+//'`
 			  done
 
 			  true
-- 
2.43.7


